src/group_theory/nilpotent.lean

```diff
@@ -460,19 +488,80 @@ begin
   exact mem_center_iff.mp (h hy1) z,
 end
 
+/-- The preimage of a nilpotent group is nilpotent if the kernel of the homomorphism is contained
+in the center -/
 lemma is_nilpotent_of_ker_le_center {H : Type*} [group H] {f : G →* H}
   (hf1 : f.ker ≤ center G) (hH : is_nilpotent H) : is_nilpotent G :=
 begin
   rw nilpotent_iff_lower_central_series at *,
   rcases hH with ⟨n, hn⟩,
-  refine ⟨n + 1, lower_central_series_succ_eq_bot
-    (le_trans ((map_eq_bot_iff _).mp _) hf1)⟩,
+  use (n + 1),
+  refine lower_central_series_succ_eq_bot (le_trans ((map_eq_bot_iff _).mp _) hf1),
   exact eq_bot_iff.mpr (hn ▸ (lower_central_series.map f n)),
 end
 
+section classical
+
+open_locale classical
+
+lemma nilpotency_class_le_of_ker_le_center {H : Type*} [group H] {f : G →* H}
+  (hf1 : f.ker ≤ center G) (hH : is_nilpotent H) :
+  @group.nilpotency_class G _ (is_nilpotent_of_ker_le_center hf1 hH) ≤
+    group.nilpotency_class H + 1 :=
+begin
+  rw ← lower_central_series_length_eq_nilpotency_class,
+  apply nat.find_min',
+  refine lower_central_series_succ_eq_bot (le_trans ((map_eq_bot_iff _).mp _) hf1),
+  apply eq_bot_iff.mpr,
+  apply (le_trans (lower_central_series.map f _)),
+  simp only [lower_central_series_nilpotency_class, le_bot_iff],
+end
+
+end classical
+
+/-- The range of a surejctive homomorphism from a nilpotent group is nilpotent -/
+lemma nilpotent_of_surjective
+  {G' : Type*} [group G'] {f : G →* G'} (hf : function.surjective f) [h : is_nilpotent G] :
```

Comment:
```suggestion
lemma nilpotent_of_surjective {G' : Type*} [group G'] [h : is_nilpotent G]
  (f : G →* G') (hf : function.surjective f) :
```
I think it's better to make `f` explicit, even though `hf` determines it. Because I think it will happen quite often in practice that people will have `f` lying around, and want `hf` as a new goal.